sormas-ui/src/main/java/de/symeda/sormas/ui/events/eventParticipantMerge/EventParticipantPickLeadEventParticipant.java

```diff
@@ -0,0 +1,244 @@
+			Set<EventParticipantSelectionDto> selectedRow = eventParticipantDuplicateGrid.getEventParticipantMergeSelectionGrid().getSelectedItems();
+			if (!selectedRow.isEmpty()) {
+				pickedEventParticipants.add(selectedRow.iterator().next().getEventParticipantUuid());
+			}
+		});
+
+		return pickedEventParticipants;
+	}
+
+	protected void initializeGrid() {
+
+		eventParticipantSelectionDtos = transformDuplicateEventParticipants(eventParticipantDtos);
+
+		eventParticipantSelectionDtos.forEach((key, value) -> {
+			List<EventParticipantSelectionDto> eventParticipantSelectionDtos =
+				mapEventParticipantToSelection(value.getDuplicateEventParticipantList());
+			EventParticipantMergeSelectionGrid newGrid = new EventParticipantMergeSelectionGrid(eventParticipantSelectionDtos);
+
+			EventSelectionDto eventSelectionDto =
+				new EventSelectionDto(eventParticipantSelectionDtos.get(0).getEventUuid(), eventParticipantSelectionDtos.get(0).getEventTitle());
+			Component eventComponent = buildEventComponent(eventSelectionDto);
+
+			EventParticipantDuplicateGrid eventParticipantDuplicateGrid = new EventParticipantDuplicateGrid();
+			eventParticipantDuplicateGrid.setEventTitle(eventComponent);
+			eventParticipantDuplicateGrid.setEventParticipantMergeSelectionGrid(newGrid);
+
+			eventParticipantMergeSelectionGrid.put(key, eventParticipantDuplicateGrid);
+
+		});
+	}
+
+	private Map<String, EventParticipantDuplicate> transformDuplicateEventParticipants(List<EventParticipantDto> duplicateEventParticipantList) {
+
+		Map<String, EventParticipantDuplicate> duplicateEvParticipantByEvent = new HashMap<>();
+
+		duplicateEventParticipantList.forEach(eventParticipantDto -> {
+
+			if (duplicateEvParticipantByEvent.containsKey(eventParticipantDto.getEvent().getUuid())) {
+				duplicateEvParticipantByEvent.get(eventParticipantDto.getEvent().getUuid())
+					.getDuplicateEventParticipantList()
+					.add(eventParticipantDto);
+			} else {
+				EventParticipantDuplicate newEventWithDuplicate = new EventParticipantDuplicate();
+				List<EventParticipantDto> newDuplicateEventParticipantList = new ArrayList<>();
+				newDuplicateEventParticipantList.add(eventParticipantDto);
+				newEventWithDuplicate.setDuplicateEventParticipantList(newDuplicateEventParticipantList);
+				duplicateEvParticipantByEvent.put(eventParticipantDto.getEvent().getUuid(), newEventWithDuplicate);
+			}
+
+		});
+
+		return duplicateEvParticipantByEvent;
+	}
+
+	public static class EventParticipantDuplicate {
+
+		private List<EventParticipantDto> duplicateEventParticipantList;
+
+		public List<EventParticipantDto> getDuplicateEventParticipantList() {
+			return duplicateEventParticipantList;
+		}
+
+		public void setDuplicateEventParticipantList(List<EventParticipantDto> duplicateEventParticipantList) {
+			this.duplicateEventParticipantList = duplicateEventParticipantList;
+		}
+	}
+
+	public static class EventParticipantDuplicateGrid {
```

Comment:
this is not a grid, please give a better name

also this class should be private

Reply:
renamed and changed